lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -163,49 +163,36 @@ def patch_process_action
         else
           # Rewrite module that gets composed into the Rails controller base class
           ::ActionController::Instrumentation.class_eval do
-            def process_action_with_datadog(*args)
-              # mutable payload with a tracing context that is used in two different
-              # signals; it propagates the request span so that it can be finished
-              # no matter what
-              payload = {
-                controller: self.class,
-                action: action_name,
-                headers: {
-                  # The exception this controller was given in the request,
-                  # which is typical if the controller is configured to handle exceptions.
-                  request_exception: request.headers['action_dispatch.exception']
-                },
-                tracing_context: {}
-              }
-
-              begin
-                # process and catch request exceptions
-                Datadog::Contrib::Rails::ActionController.start_processing(payload)
-                result = process_action_without_datadog(*args)
-                payload[:status] = response.status
-                result
-              rescue Exception => e
-                payload[:exception] = [e.class.name, e.message]
-                payload[:exception_object] = e
-                raise e
-              end
-            ensure
-              Datadog::Contrib::Rails::ActionController.finish_processing(payload)
-            end
-
             alias_method :process_action_without_datadog, :process_action
-            alias_method :process_action, :process_action_with_datadog
+            remove_method :process_action
+
+            include ActionControllerPatch
           end
         end
       end
     end
 
+    # Compat module for Ruby versions not supporting #prepend
+    module CompatActionControllerPatch
+      def process_action(*args)
+        process_action_without_datadog(*args)
+      end
+    end
+
     # ActionController patch for Ruby 2.0+
     module ActionControllerPatch
+      # compat for Ruby versions not supporting #prepend
+      include CompatActionControllerPatch unless Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.0.0')
```

Comment:
How does this line work? Won't doing an `include` here only include it into the `ActionControllerPatch`, and not into the `ActionController::Instrumentation` module? Typically the only way to nest module inheritance is to implement the `def self.included(base)` method.

Reply:
Its here to build proper ancestry chain for ActionControllerPatch to work when included in Ruby < 2.0.